### PR TITLE
Extend `ObjectChange` `change_context_detail` to 400 chars

### PR DESCRIPTION
**Pull request by chadell**
# Closes: https://github.com/nautobot/nautobot-plugin-circuit-maintenance/issues/218
# What's Changed

- The `change_context_detail` added in version 1.4.1 was limited to 100 chars
- Some applications have very long names (>100), as the one referenced in the issue: plugins-nautobot_circuit_maintenance-handle_notifications-handler-handlecircuitmaintenancenotifications
- The fix increases `change_context_detail` from 100 to 400 chars

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by chadell** [comment]
Adding @gsnider2195 as the original code contributor

**Comment by jvanderaa** [comment]
Confirms to fix https://github.com/nautobot/nautobot-plugin-circuit-maintenance/issues/218

**Comment by glennmatthews** [comment]
Can we also make this more robust in general so that we don't hit this same issue in the future if (heaven forbid) we somehow end up with a plugin with a 200-character viewname? It'd be good to ensure that we automatically truncate the change_context_detail to fit, e.g. here at least:

https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/signals.py#L83

File: changes/2602.changed [hunk 1]
```diff
@@ -0,0 +1 @@
+`ObjectChange.change_context_detail` extended from 100 to 200 chars
```
Comment (glennmatthews) [inline changes/2602.changed:1]:
```suggestion
Increased size of `ObjectChange.change_context_detail` field from 100 to 200 chars.
```

**Review by glennmatthews — COMMENTED** [review]
Come to think of it, we have a similar potential issue with `object_repr` - it would be good to add a length check and automatic truncation [here](https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/models/change_logging.py#L40) as well.

**Comment by glennmatthews** [comment]
Ah - this is actually coming from https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/jobs.py#L1243. Since Jobs can have a slug [up to 320 characters in the worst case](https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/constants.py#L39), we definitely need some truncation logic and/or to increase the limit to more than 200 chars.

**Comment by chadell** [comment]
> https://github.com/nautobot/nautobot/blob/develop/nautobot/extras/jobs.py#L1243

I agree on the idea. I would cover the `object_repr` too.
I believe that the truncation is the most conservative approach, as it will cover our backs whatever we find in the future.
If there are no concerns, I would apply a 200 to both attributes, with truncation.

**Comment by jvanderaa** [comment]
@chadell can we move to a 320 size on the context to match that of a slug?

**Comment by chadell** [comment]
We could move to 400 (to account for the application name). 
But, I will also apply truncation to solve this once and for all.

**Comment by bryanculver** [comment]
@chadell Moved to draft to indicate you're still updating the PR to make it visually distinct.

Please select "Ready for review" once the PR has been updated.

**Review by glennmatthews — COMMENTED** [review]
Looks functionally solid to me. Can we define and use some constants in `nautobot.extras.constants` for these two values so that we don't have to have hard-coded `200` and `400` in various places?

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**